controller/Dockerfile

```diff
@@ -1,9 +1,27 @@
-ARG OS_VERSION=ltsc2019
+ARG OS_VERSION
+
+# pinned base images
+
+# mcr.microsoft.com/oss/go/microsoft/golang:1.22.3-1-cbl-mariner2.0
+FROM --platform=$BUILDPLATFORM mcr.microsoft.com/oss/go/microsoft/golang@sha256:8253def0216b87b2994b7ad689aeec7440f6eb67f981e438071d8d67e36ff69f as golang
```

Comment:
do we really need to pin the Go image? 🤔

Reply:
For supply chain security, we should pin all dependencies. For Go code we get this for free with Go modules; for containers, that means referencing them by the SHA like this.